chained/proxy.go

```diff
@@ -212,7 +212,7 @@ func createImpl(configDir, name, addr, transport string, s *config.ProxyConfig,
 	case "algeneva":
 		impl, err = newAlgenevaImpl(addr, s, reportDialCore)
 	case "water":
-		impl, err = newWaterImpl(addr, s, reportDialCore)
+		impl, err = newWaterImpl(configDir, addr, s, reportDialCore)
```

Comment:
Looks like this will now populate the config directory with .wasm and .last-loaded files.  I think we still want a sub-directory for those files.  You asked [here](https://github.com/getlantern/flashlight/pull/1427#discussion_r1815590387) where this directory should be specified and created.  I think this is an appropriate location.  How about something like:
```
waterDir := filepath.Join(configDir, "water")
if err := os.Mkdir(waterDir, 0755); err != nil && !errors.Is(err, fs.ErrExist) {
    return nil, fmt.Errorf("failed to create water directory: %w", err)
}
impl, err = newWaterImpl(waterDir, addr, s, reportDialCore)
```

Reply:
Just FYI `err != nil` isn't necessary, `errors.Is` checks for `nil`.

Reply:
Also, `os.MkdirAll` won't return an error if the directory already exists. Either way, you could drop `err != nil` or `!errors.Is(err, fs.ErrExist)` if you use `os.MkdirAll`. This is just an FYI!

Reply:
Moving to `os.MkdirAll`, thanks for the suggestion!

Reply:
Oh maybe you're thinking of `os.MkdirAll`.  That won't error if the directory already exists.  You should use that @WendelHime.  Then you can use @garmr-ulfr's suggested simplification.

Reply:
Wait I'm a dummy.  `os.MkdirAll` is what you suggested @garmr-ulfr.

Reading comprehension... Sorry for the confusion 😬 🤦

Reply:
> > Just FYI err != nil isn't necessary, errors.Is checks for nil.
> 
> The check is for the negation. `!errors.Is(err, fs.ErrExist)` on its own will miss the case in which we have a non-nil error due to something other than `fs.ErrExist`.

You're correct. `errors.Is(err, fs.ErrExist)` would return false because `nil != fs.ErrExist`. Ignore that part of what I said. 

> Wait I'm a dummy. `os.MkdirAll` is what you suggested @garmr-ulfr.
> 
> Reading comprehension... Sorry for the confusion 😬 🤦

:laughing: I had my own "dummy" moment. We'll just call it even. It's Friday.